.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,49 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/cpp
+{
+    "name": "C++",
+    "build": {
+        "dockerfile": "Dockerfile",
+        // Update 'VARIANT' to pick an Debian / Ubuntu OS version: debian-11, debian-10, ubuntu-22.04, ubuntu-20.04, ubuntu-18.04
+        // Use Debian 11, Ubuntu 18.04 or Ubuntu 22.04 on local arm64/Apple Silicon
+        "args": {
+            "VARIANT": "ubuntu-22.04",
+            "REINSTALL_CMAKE_VERSION_FROM_SOURCE": "3.24.2"
```

Comment:
> Wouldn't everything be clearer if Dockerfile had the right options instead of being meaningless and overriden by the options in this file?

If this version was hardcoded then (somewhat obviously) you would not be able to change the CMake version in the image, which sort of plays into your second question:

> Also, I'm curious why you need to "reinstall cmake from source" - what's wrong with whatever version of cmake version Ubuntu 22.04 has?

Technically as of the current state of the code, nothing. But say that next year, Seastar adds a CMake change that depends on recent CMake feature. In that case, I figured it would be better to future-proof the change and make it user-configurable + obvious how to set the version.

I'm not married to the idea, happy to revert it if you think something like this will never happen (or just set `REINSTALL_CMAKE_VERSION_FROM_SOURCE=none`)